ecosystem/typescript/sdk_v2/src/crypto/multi_ed25519.ts

```diff
@@ -0,0 +1,193 @@
+   * @param args.bitmap 4 bytes, at most 32 signatures are supported. If Nth bit value is `1`, the Nth
+   * signature should be provided in `signatures`. Bits are read from left to right
+   */
+  constructor(args: { signatures: Signature[]; bitmap: Uint8Array }) {
+    const { signatures, bitmap } = args;
+    if (bitmap.length !== MultiSignature.BITMAP_LEN) {
+      throw new Error(`"bitmap" length should be ${MultiSignature.BITMAP_LEN}`);
+    }
+
+    this.signatures = signatures;
+    this.bitmap = bitmap;
+  }
+
+  /**
+   * Converts a MultiSignature into Uint8Array (bytes) with `bytes = s1_bytes | ... | sn_bytes | bitmap`
+   */
+  toUint8Array(): Uint8Array {
+    const bytes = new Uint8Array(this.signatures.length * Signature.LENGTH + MultiSignature.BITMAP_LEN);
+    this.signatures.forEach((k: Signature, i: number) => {
+      bytes.set(k.toUint8Array(), i * Signature.LENGTH);
+    });
+
+    bytes.set(this.bitmap, this.signatures.length * Signature.LENGTH);
+
+    return bytes;
+  }
+
+  /**
+   * Helper method to create a bitmap out of the specified bit positions
+   * @param bits The bitmap positions that should be set. A position starts at index 0.
+   * Valid position should range between 0 and 31.
+   * @example
+   * Here's an example of valid `bits`
+   * ```
+   * [0, 2, 31]
+   * ```
+   * `[0, 2, 31]` means the 1st, 3rd and 32nd bits should be set in the bitmap.
+   * The result bitmap should be 0b1010000000000000000000000000001
+   *
+   * @returns bitmap that is 32bit long
+   */
+  static createBitmap(bits: number[]): Uint8Array {
+    // Bits are read from left to right. e.g. 0b10000000 represents the first bit is set in one byte.
+    // The decimal value of 0b10000000 is 128.
+    const firstBitInByte = 128;
+    const bitmap = new Uint8Array([0, 0, 0, 0]);
+
+    // Check if duplicates exist in bits
+    const dupCheckSet = new Set();
+
+    bits.forEach((bit: number) => {
+      if (bit >= MultiSignature.MAX_SIGNATURES_SUPPORTED) {
+        throw new Error(`Invalid bit value ${bit}.`);
+      }
+
+      if (dupCheckSet.has(bit)) {
+        throw new Error("Duplicated bits detected.");
+      }
+
+      dupCheckSet.add(bit);
+
+      const byteOffset = Math.floor(bit / 8);
+
+      let byte = bitmap[byteOffset];
+
+      byte |= firstBitInByte >> bit % 8;
+
+      bitmap[byteOffset] = byte;
+    });
+
+    return bitmap;
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.toUint8Array());
+  }
+
+  static deserialize(deserializer: Deserializer): MultiSignature {
+    const bytes = deserializer.deserializeBytes();
+    const bitmap = bytes.subarray(bytes.length - 4);
+
+    const sigs: Signature[] = [];
+
+    for (let i = 0; i < bytes.length - bitmap.length; i += Signature.LENGTH) {
+      const begin = i;
+      sigs.push(new Signature({ data: bytes.subarray(begin, begin + Signature.LENGTH) }));
+    }
+    return new MultiSignature({ signatures: sigs, bitmap });
```

Comment:
nit - change `sigs` to `signatures` and then can
```suggestion
    return new MultiSignature({ signatures, bitmap });
```